tests/lib/tools/snaps-state

```diff
@@ -22,6 +22,11 @@ pack_local() {
     local SNAP_DIR="${2:-$TESTSLIB/snaps/${SNAP_NAME}}"
     local SNAP_VERSION="${3:-1.0}"
 
+    # Use local snap in case it exists
+    if [ -d "$SNAP_NAME" ]; then
+        SNAP_DIR="$PWD/$SNAP_NAME"
+    fi
```

Comment:
Should we fail if same snap exists both in local and $TESTSLIB/snaps directory? IMHO that would help clear any confusion and mistakes.

Reply:
yes, you are right, that check is needed, thanks for the suggestion

Reply:
The check you added is fine, but I was suggesting something else: to error out if same snap is defined twice (in local dir and in TESTLIB); this would help avoid confusion about what snap is actually used by the test.

Reply:
ah, got it, I'll add that as well

Reply:
Hmmm, do you mean "$PWD/$SNAP_NAME" and "$TESTLIB/$SNAP_NAME" could be the same? You could chck if "$TESTLIB" != "$PWD" first and only then do a check for duplicated name under both, but feel free to consider this as a followup, it's not so important for this PR.
A check inside install_local sounds good.